py3status/modules/online_status.py

```diff
@@ -53,23 +47,26 @@ class Meta:
                     'msg': 'obsolete parameter use `icon_off`',
                 },
             ],
+            'remove': [
+                {
+                    'param': 'timeout',
+                    'msg': 'obsolete parameter',
+                },
+            ]
         }
 
+    def post_config_hook(self):
+        self.color_on = self.py3.COLOR_ON or self.py3.COLOR_GOOD
+        self.color_off = self.py3.COLOR_OFF or self.py3.COLOR_BAD
+
     def _connection_present(self):
-        if '://' in self.url:
-            try:
-                urlopen(self.url, timeout=self.timeout)
-            except:
-                return False
-            else:
-                return True
+        try:
+            self.url = urlsplit(self.url).netloc
```

Comment:
you probably only want to do this once so in post_config_hook() and only if we have `://` in the string